applications/operator_sdk/osdk-leader-election.adoc

```diff
@@ -0,0 +1,41 @@
+[id="osdk-leader-election"]
+= Configuring leader election
+include::modules/common-attributes.adoc[]
+:context: osdk-leader-election
+
+toc::[]
+
+During the lifecycle of an Operator, it is possible that there may be more than
+one instance running at any given time, for example when rolling out an upgrade
+for the Operator. In such a scenario, it is necessary to avoid contention
+between multiple Operator instances using leader election. This ensures only one
+leader instance handles the reconciliation while the other instances are
+inactive but ready to take over when the leader steps down.
+
+There are two different leader election implementations to choose from, each
+with its own trade-off:
+
+* _Leader-for-life_: The leader Pod only gives up leadership (using garbage
+collection) when it is deleted. This implementation precludes the possibility of
+two instances mistakenly running as leaders (split brain). However, this method
+can be subject to a delay in electing a new leader. For example, when the leader
+Pod is on an unresponsive or partitioned node, the
+link:https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/#options[`pod-eviction-timeout`]
+dictates how it takes for the leader Pod to be deleted from the node and step
+down (default `5m`). See the link:https://godoc.org/github.com/operator-framework/operator-sdk/pkg/leader[Leader-for-life] Go documentation for more.
+
+* _Leader-with-lease_: The leader Pod periodically renews the leader lease and
+gives up leadership when it cannot renew the lease. This implementation allows
+for a faster transition to a new leader when the existing leader is isolated,
+but there is a possibility of split brain in
+link:https://github.com/kubernetes/client-go/blob/30b06a83d67458700a5378239df6b96948cb9160/tools/leaderelection/leaderelection.go#L21-L24[certain situations]. See the
+link:https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/leaderelection[Leader-with-lease]
+Go documentation for more.
+
+By default, the Operator SDK enables the Leader-for-life implementation. Consult
+the related Go documentation for both approaches to consider the trade-offs that make
+sense for your use case,
+
+The following examples illustrate how to use the two options.
+
+include::modules/osdk-leader-election-types.adoc[leveloffset=+1]
```

Comment:
Error link, I cannot open it.

Reply:
This isn't a link, it just includes the `osdk-leader-election-types.adoc` module in place in the rendered version:

http://file.rdu.redhat.com/~adellape/053019/leaderelection/applications/operator_sdk/osdk-leader-election.html#osdk-leader-for-life-election_osdk-leader-election

Reply:
ok, got it, tthanks!